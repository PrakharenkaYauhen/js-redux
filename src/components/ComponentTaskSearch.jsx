// ComponentTaskSearch

import React from 'react';
import PropTypes from 'prop-types';

function ComponentTaskSearch({
    searchInputValue,
    currentlyTaskList,
    itemClick,
    onSearchChange,
}) {

    let currentlyTaskListDOM = [];

    if (currentlyTaskList) {
        currentlyTaskListDOM = currentlyTaskList.filter(item =>
            item.content.toLowerCase().indexOf(searchInputValue.toLowerCase()) !== -1 && searchInputValue.length !== 0)
            .map((item, i) =>
                <li className='search__item' key={i} onClick={itemClick}>{item.content}</li>)
    }

    let choiceList = () => {
        return (currentlyTaskListDOM.length !== 0) && (
            <div className='search__items'>
                <ul className='search__list'>
                    {currentlyTaskListDOM}
                </ul>
            </div>
        )
    }

    return (
        <div className='search'>
            <label className='search__label'>
                <span className='search__span'>Enter a serching task:</span>
                <input className='search__input'
                    type="search"
                    value={searchInputValue}
                    onChange={onSearchChange}
                />
                {choiceList()}
            </label>
        </div>
    )
}

ComponentTaskSearch.propTypes = {
    searchInputValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    currentlyTaskList: PropTypes.array,
    itemClick: PropTypes.func,
    onSearchChange: PropTypes.func,
}

export { ComponentTaskSearch };