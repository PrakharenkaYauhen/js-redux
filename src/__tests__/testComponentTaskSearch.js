// testComponentTaskSearch

import React from 'react';
import ComponentTaskSearch from '../components/ComponentTaskSearch';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const searchBlock = renderer
  .create(<ComponentTaskSearch/>)
  .toJSON();
  expect(searchBlock).toMatchSnapshot();  
});