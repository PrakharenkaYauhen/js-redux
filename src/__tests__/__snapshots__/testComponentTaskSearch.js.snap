// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="search"
>
  <label
    className="search__label"
    htmlFor="search__input"
  >
    <span
      className="search__span"
    >
      Enter a serching task:
    </span>
    <input
      className="search__input"
      id="search__input"
      tabIndex="50"
      type="search"
      value=""
    />
  </label>
</div>
`;
